cmd/goose/main.go

```diff
@@ -42,6 +44,10 @@ var (
 var version string
 
 func main() {
+	if ok, err := strconv.ParseBool(os.Getenv("GOOSE_CLI")); err == nil && ok {
```

Comment:
I'd like to avoid a long-running feature branch, so this gate is to allow us to iterate and also use it in real-world projects.